src/main/java/edu/kit/provideq/toolbox/GamsProcessRunner.java

```diff
@@ -27,19 +28,30 @@ public class GamsProcessRunner extends ProcessRunner {
    */
   private static final String GAMS_EXECUTABLE_NAME = "gams";
 
+  /**
+   * Creates a process runner for a GAMS task.
+   * @param directory the working directory to run GAMS in.
+   * @param scriptFileName the filename of the GAMS script to run.
+   */
+  public GamsProcessRunner(String directory, String scriptFileName) {
+    this(directory, scriptFileName, new String[0]);
+  }
+
   /**
    * Creates a process runner for a GAMS task.
    * @param directory the working directory to run GAMS in.
    * @param scriptFileName the filename of the GAMS script to run.
    * @param arguments extra arguments to pass to GAMS. Use this to pass problem input to the solver.
    */
-  public GamsProcessRunner(File directory, String scriptFileName, String... arguments) {
-    super(buildGamsProcess(directory, scriptFileName, arguments));
+  public GamsProcessRunner(String directory, String scriptFileName, String... arguments) {
+    super(createGenericProcessBuilder(directory, GAMS_EXECUTABLE_NAME, scriptFileName, arguments));
+
+    addProblemFilePathToProcessCommand("--INPUT=\"%s\"");
   }
 
   @Override
-  public ProcessResult run() throws IOException, InterruptedException {
-    var result = super.run();
+  public ProcessResult run(ProblemType problemType, long solutionId, String problemData) {
```

Comment:
Why are some arguments in the constructor and some in the run method? I thought the `GamsProcessRunner` isn't reusable?

Reply:
This follows the way of `ProcessRunner`. Do you mean to say that `ProcessRunner` shouldn't be using these arguments?

Reply:
Technically the parts passed in ProcessResult are reusable, so passing different values into the `run` method can end up with different results. Passing them in the constructor would make sense in that sense, but it has to be done for `ProcessRunner` too. It should be noted that having this will increase the arguments that need to be passed into the bean, which won't be enforced by the compiler or the IDE and may lead to runtime exceptions.

Reply:
> Passing them in the constructor would make sense in that sense, but it has to be done for ProcessRunner too.
That sounds reasonable.

> It should be noted that having this will increase the arguments that need to be passed into the bean, which won't be enforced by the compiler or the IDE and may lead to runtime exceptions.
Ohhh, I didn't see that. Why don't we call the constructor directly?

Reply:
We have this bit which needs to be autowired by Spring, otherwise we could call the constructor directly too.
```java
    @Autowired
    public void setResourceProvider(ResourceProvider resourceProvider) {
        this.resourceProvider = resourceProvider;
    }
```
`ResourceProvider` is used to get the problem directory path.

Do you see a way to do this with a constructor only?